### PR TITLE
Hoist version string to namespace level

### PR DESCRIPTION
**Pull request by nbougalis**

### CODE REVIEW

**Comment by vinniefalco** [comment]
Why the raw string and not the `std::string`?

File: src/ripple/protocol/impl/BuildInfo.cpp [hunk 1]
```diff
@@ -27,21 +27,17 @@ namespace ripple {
 
 namespace BuildInfo {
 
-char const* getRawVersionString ()
-{
-    static char const* const rawText =
+//------------------------------------------------------------------------------
+char const* const versionString =
```
Comment (HowardHinnant) [inline src/ripple/protocol/impl/BuildInfo.cpp]: I recommend:
````
constexpr char const* const versionString =
````
This gives `versionString` internal linkage (file scope).  After experimenting with online compilers, I believe this incantation will satisfy all 3 of our compilers with no warnings.

**Comment by nbougalis** [comment]
I wanted as low-level a type as possible: a pointer into the read-only segment of the executable will _always_ be accessible from all contexts, including a core file. A `std::string` may not.

**Comment by HowardHinnant** [comment]
> Why the raw string and not the std::string?

If we use `std::string` at file scope, that means we need to construct that `std::string` at run time prior to `main`.  This isn't a huge deal, but the fewer run time initializations we have the better.  The raw string will be laid down in ROM at compile time.

We then construct a `std::string` from the raw string exactly once when the version is first actually requested at run time (lazy initialization).

File: src/ripple/protocol/impl/BuildInfo.cpp (hunk 1 above)
Comment (HowardHinnant) [inline src/ripple/protocol/impl/BuildInfo.cpp]: Oh, except that I guess your intention is for `versionString` to have extern linkage! :-)  I missed that on first read.  How do we grab `versionString` for the email report?

Comment (nbougalis) [inline src/ripple/protocol/impl/BuildInfo.cpp]: Augment the `gdb` script to execute:

`print 'ripple::BuildInfo::versionString'`

Comment (HowardHinnant) [inline src/ripple/protocol/impl/BuildInfo.cpp]: Sounds like a plan, thanks.

**Comment by HowardHinnant** [comment]
👍

**Comment by nbougalis** [comment]
Not sure what's up with the coverage build failure - I've restarted it twice and it's failed twice.

**Comment by nbougalis** [comment]
Merged as 08ea2451013052fbeeb9ed2ccf894cc56668dfd8

**Pull request closed without merging**